test/e2e/framework/providers/azure/azure.go

```diff
@@ -41,8 +41,9 @@ func newProvider() (framework.ProviderInterface, error) {
 	if err != nil {
```

Comment:
should not return directly an error?
```
return nil, err
```
it doesn't seem like it can create a valid azure provider

Reply:
The nil config will be handle in `azure.NewCloud`, Azure cloud provider would be initialized from secret